types/plotly.js/index.d.ts

```diff
@@ -85,21 +85,19 @@ export type PlotRestyleEvent = [
 	number[]	// array of traces updated
 ];
 
-export interface PlotAxis {
-	range: [number, number];
-	autorange: boolean;
-}
-
 export interface PlotScene {
 	center: Point;
 	eye: Point;
 	up: Point;
 }
 
 export interface PlotRelayoutEvent {
```

Comment:
When a plot is resized, it appears that the PlotRelayoutEvent contains the Layout information which is not included in this change.

Additionally, all the properties that were just added to the PlotRelayoutEvent interface should be marked as optional since not all of them are expected to be returned.  If you only zoom in horizontally, then the yaxis.* properties will not be included.  Similarly, if you use the toolbar button to reset the zoom level, then only the xaxis.autorange and yaxis.autorange values will be returned.

Therefore I believe the correct interface definition should be:

```
export interface PlotRelayoutEvent extends Partial<Layout> {
	'xaxis.range[0]'? : number; 
	'xaxis.range[1]'? : number; 
	'yaxis.range[0]'? : number; 
	'yaxis.range[1]'? : number; 
	'xaxis.autorange'? : boolean; 
	'yaxis.autorange'? : boolean;
}
```

Reply:
@alexcjohnson

Reply:
@etpinard

Reply:
good point about optional, completely forgot about them, though I've used Codepen sample to catch differences at runtime. To be amended

Reply:
So the event props are now optional after amending commit

Reply:
> When a plot is resized, it appears that the PlotRelayoutEvent contains the Layout information which is not included in this change

I haven't seen that in the CodePen sample, but I've amended the PR to incorporate that change too. 
Thanks!